src/encoder.c

```diff
@@ -93,24 +113,31 @@ uint16_t get_encoder_right_total_count(void)
  */
 void update_encoder_readings(void)
 {
-	static uint16_t left_latest_count;
-	static uint16_t right_latest_count;
+	static uint16_t last_left_count;
+	static uint16_t last_right_count;
 
 	uint16_t left_count;
 	uint16_t right_count;
 
+	/* Encoder counters */
 	left_count = read_encoder_left();
 	right_count = read_encoder_right();
-
-	left_diff_count = (int32_t)(left_count - left_latest_count);
-	right_diff_count = (int32_t)(right_count - right_latest_count);
-
 	left_total_count += left_diff_count;
 	right_total_count += right_diff_count;
-
-	left_distance = left_total_count * MILLIMETERS_PER_COUNT;
-	right_distance = right_total_count * MILLIMETERS_PER_COUNT;
-
-	left_latest_count = left_count;
-	right_latest_count = right_count;
+	left_diff_count = (int32_t)(left_count - last_left_count);
+	right_diff_count = (int32_t)(right_count - last_right_count);
+
+	/* Distance */
+	left_micrometers = (int32_t)(left_total_count * MICROMETERS_PER_COUNT);
+	right_micrometers =
+	    (int32_t)(right_total_count * MICROMETERS_PER_COUNT);
+
+	/* Speed */
+	left_speed = left_diff_count * (MICROMETERS_PER_COUNT / 1000000.) *
```

Comment:
Change the 1000000. for a macro.I would multiply by SYSTICK_FREQUENCY first, (I'm not sure but I think that like that one of the operation is with integer instead of float)

Reply:
Added a new macro for the `1000000.`. The order in the multiplication does not really matter, as the compiler should optimize that for us (that way we keep the "natural" order of the equation).